packages/checkout/utils/validation/isPostcode.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { POSTCODE_REGEXES } from 'postcode-validator/lib/cjs/postcode-regexes.js';
+
+const CUSTOM_REGEXES = new Map< string, RegExp >( [
+	[ 'BA', /^([7-8]{1})([0-9]{4})$/ ],
+	[
+		'GB',
+		/^([A-Z]){1}([0-9]{1,2}|[A-Z][0-9][A-Z]|[A-Z][0-9]{2}|[A-Z][0-9]|[0-9][A-Z]){1}([ ])?([0-9][A-z]{2}){1}|BFPO(?:\s)?([0-9]{1,4})$|BFPO(c\/o[0-9]{1,3})$/i,
+	],
+	[ 'IN', /^[1-9]{1}[0-9]{2}\s{0,1}[0-9]{3}$/ ],
```

Comment:
Can we change the regex for Indian postcodes?  The Indian postcodes don't contain spaces. They are 6-digit numeric codes without space, letters, and special characters.

Reply:
Thanks for your review, @tarunvijwani. You are correct that Indian postcodes technically do not contain spaces. The initial regex for Indian postcodes of WooCommerce core did not allow spaces: https://github.com/woocommerce/woocommerce/commit/4fc66caadb3c82f1e4b4db541bed9a33737182ae

In https://github.com/woocommerce/woocommerce/pull/27546, the regex had been changed to allow spaces. In https://github.com/woocommerce/woocommerce/pull/27546#discussion_r483066262, @claudiosanches mentioned:
> This is for validation, while some system doesn't use space, it still is an allowed format, we shouldn't throw warnings if someone enters a space, that's why the regex needs to allow it, making more user friendly, if not will be very common for people getting frustrated by entering a valid zip code like 282 002 and getting a message that is invalid.

While you're correct in pointing this out, @tarunvijwani, it appears that some folks in India are using a space between the first and the last three digits. I agree with @claudiosanches that we should aim for a user-friendly system, even if this means that we slightly divert from the official rules and also support common practices, that technically violate the postcode schema. That said, I'm keen in hearing your thoughts on this.